dm/master/openapi.go

```diff
@@ -312,3 +376,201 @@ func modelToSourceCfg(source openapi.Source) *config.SourceConfig {
+			if fullCfg.ExportThreads != nil {
+				subTaskCfg.MydumperConfig.Threads = *fullCfg.ExportThreads
+			}
+			if fullCfg.DataDir != nil {
+				subTaskCfg.LoaderConfig.Dir = *fullCfg.DataDir
+			}
+		}
+		// set incremental config
+		subTaskCfg.SyncerConfig = config.DefaultSyncerConfig()
+		if incrCfg := task.SourceConfig.IncrMigrateConf; incrCfg != nil {
+			if incrCfg.ReplThreads != nil {
+				subTaskCfg.SyncerConfig.WorkerCount = *incrCfg.ReplThreads
+			}
+			if incrCfg.ReplBatch != nil {
+				subTaskCfg.SyncerConfig.Batch = *incrCfg.ReplBatch
+			}
+		}
+		// set route , blockAllowList, filter config
+		doDBs := []string{}
+		doTables := []*filter.Table{}
+		routeRules := []*router.TableRule{}
+		filterRules := []*bf.BinlogEventRule{}
+		for _, rule := range tableMigrateRuleMap[sourceCfg.SourceName] {
+			// route
+			routeRules = append(routeRules, &router.TableRule{
+				SchemaPattern: rule.Source.Schema,
+				TablePattern:  rule.Source.Table,
+				TargetSchema:  rule.Target.Schema,
+				TargetTable:   rule.Target.Table,
+			})
+			// filter
+			if rule.EventFilterName != nil {
+				for _, name := range *rule.EventFilterName {
+					filterRule, ok := eventFilterTemplateMap[name] // NOTE: this return a cpoied value
+					if !ok {
+						return nil, terror.ErrOpenAPICommonError.Generatef("filter rule name=%s not found", name)
+					}
+					filterRule.SchemaPattern = rule.Source.Schema
+					filterRule.TablePattern = rule.Source.Table
+					filterRules = append(filterRules, &filterRule)
+				}
+			}
+			// BlockAllowList
+			doDBs = append(doDBs, rule.Source.Schema)
+			doTables = append(doTables, &filter.Table{
+				Schema: rule.Source.Schema,
+				Name:   rule.Source.Table,
+			})
+		}
+		subTaskCfg.RouteRules = routeRules
+		subTaskCfg.FilterRules = filterRules
+		subTaskCfg.BAList = &filter.Rules{DoDBs: doDBs, DoTables: doTables}
+		// adjust sub task config
+		if err := subTaskCfg.Adjust(true); err != nil {
+			return nil, terror.Annotatef(err, "source name=%s", sourceCfg.SourceName)
+		}
+		// pre-check filter rules
+		_, err := bf.NewBinlogEvent(subTaskCfg.CaseSensitive, subTaskCfg.FilterRules)
+		if err != nil {
+			return nil, terror.ErrConfigBinlogEventFilter.Delegate(err)
+		}
+		subTaskCfgList[i] = *subTaskCfg
+	}
+	return subTaskCfgList, nil
+}
+
+func adjustModelTask(task *openapi.Task) {
+	defaultMetaSchema := "dm_meta"
+	if task.MetaSchema == nil {
+		task.MetaSchema = &defaultMetaSchema
+	}
+	// check some not implemented features
+	if task.OnDuplication != openapi.TaskOnDuplicationError {
+		log.L().Warn("now on duplication is currently only implemented at the error level")
+		task.OnDuplication = openapi.TaskOnDuplicationError
+	}
+}
+
+func getAndAdjustTaskTargetDBCfg(ctx context.Context, task *openapi.Task) (*config.DBConfig, error) {
+	toDBCfg := &config.DBConfig{
+		Host:     task.TargetConfig.Host,
+		Port:     task.TargetConfig.Port,
+		User:     task.TargetConfig.User,
+		Password: task.TargetConfig.Password,
```

Comment:
need `Session` field

Reply:
We are still missing some other configurations, which will be discussed [here](https://docs.google.com/document/d/1UYSfiF79sL9YysnTgolJrWGsP7q77q8ocFkZtpZTgrM/edit?disco=AAAAOi61M9c)